sector-builder/src/metadata_manager.rs

```diff
@@ -110,10 +110,10 @@ impl<T: KeyValueStore> SectorMetadataManager<T> {
     pub fn create_generate_post_task_proto(
         &self,
         comm_rs: &[[u8; 32]],
-        challenge_seed: &[u8; 32],
-        faults: Vec<SectorId>,
+        randomness: &[u8; 32],
+        faults: Option<Vec<SectorId>>,
```

Comment:
Opinion 1: The `Vec` makes the `Option` redundant. Just pass an empty vector instead of `None` if you want to communicate "no faults."

Opinion 2: This method is now used by the candidate-generating and PoSt-generating operations, so I think that it should probably have a name which communicates as much.